src/librustc_session/options.rs

```diff
@@ -499,6 +501,15 @@ macro_rules! options {
             }
         }
 
+        fn parse_cfguard(slote: &mut Option<CFGuard>, v: Option<&str>) -> bool {
+            match v {
+                Some("nochecks") => *slote = Some(CFGuard::Nochecks),
+                Some("checks") => *slote = Some(CFGuard::Checks),
```

Comment:
It is typical for these kinds options to also accept a value that explicitly disables the behaviour, This option should have such a value as well. In doing so you should also add an explicit `Disabled` or `No` variant to `CFGuard` (and store just `CFGuard` rather than `Option<CFGuard>`).

Reply:
Thanks for the suggestion! I've added a `CFGuard::Disabled` variant, and am now just storing `CFGuard`.